lib/DocumentManager.php

```diff
@@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
 
         return $resolver;
     }
+
+    /**
+     * If the exception is an instanceof `DocumentManagerException` set the
+     * document manager name that threw.
+     *
+     * @param \Exception $e
+     * @param string $context
+     */
+    private function processException(\Exception $exception, $context)
```

Comment:
An example I just discovered - when we encounter PHPCR exceptions we
probably want to catch them (Sulu should not know about the PHPCR
layer). As this method is here we can easily catch these exceptions in 
addition to the DocumentManager exceptions.

On Thu, Apr 28, 2016 at 12:13:23AM -0700, Daniel Rotter wrote:
>    In [1]lib/DocumentManager.php:
> 
>  > @@ -175,4 +265,34 @@ private function getOptionsResolver($eventName)
>  >
>  >          return $resolver;
>  >      }
>  > +
>  > +    /**
>  > +     * If the exception is an instanceof `DocumentManagerException` set the
>  > +     * document manager name that threw.
>  > +     *
>  > +     * @param \Exception $e
>  > +     * @param string $context
>  > +     */
>  > +    private function processException(\Exception $exception, $context)
> 
>    Why? It is a private method, so it's only called by this class. And
>    changing it the way I've suggested would have the same result, without
>    catching \Exception, which only rarely makes sense, and having "flatter"
>    code in this method.
> 
>    —
>    You are receiving this because you authored the thread.
>    Reply to this email directly or [2]view it on GitHub
> 
>    Reverse link: [3]unknown
> 
> References
> 
>    Visible links
>    1. https://github.com/sulu/sulu-document-manager/pull/66#discussion_r61382114
>    2. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61382114
>    3. https://github.com/sulu/sulu-document-manager/pull/66/files/05556aabdb564b020fa388baf338b54d39353978#r61382114